### PR TITLE
feat: retry delay period as param

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
## Description

We're having trouble getting gomock to work with throttling v2 e2e tests. This PR makes the "retry delay period" w.r.t throttling v2 a param so it can be mutated to a small value in e2e tests.

Currently, the retry delay period is a const set to 1 hour. 1 hour cannot be feasibly used for (non go-mock) e2e testing.

Pb break check fails as expected due to base branch. 

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [ ] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] Added `!` to the type prefix if state-machine breaking change (i.e., requires coordinated upgrade)
* [ ] Confirmed this PR does not introduce changes requiring state migrations, OR migration code has been added to consumer and/or provider modules
* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
* [ ] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#testing)
* [ ] Added a changelog entry to `CHANGELOG.md`
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [ ] Confirmed all CI checks have passed
* [ ] If this PR is library API breaking, bump the go.mod version string of the repo, and follow through on a new major release for both the consumer and provider

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [ ] confirmed `!` in the type prefix if API or client breaking change
* [ ] confirmed this PR does not introduce changes requiring state migrations, OR confirmed migration code has been added to consumer and/or provider modules
* [ ] confirmed all author checklist items have been addressed
* [ ] reviewed state machine logic
* [ ] reviewed API design and naming
* [ ] reviewed documentation is accurate
* [ ] reviewed tests and test coverage

### CODE REVIEW

File: x/ccv/consumer/types/params_test.go [hunk 1]
```diff
@@ -19,59 +19,63 @@ func TestValidateParams(t *testing.T) {
...
 		},
 		{
 			"custom invalid params, invalid reward denom",
-			ccvtypes.NewParams(true, 5, "", "", 5, 1005, "0.5", 1000, 24*21*time.Hour, "0.05", []string{"u"}, []string{}), false,
+			ccvtypes.NewParams(true, 5, "", "", 5, 1005, "0.5", 1000, 24*21*time.Hour, "0.05", []string{"u"}, []string{}, 2*time.Hour), false,
 		},
 		{
 			"custom invalid params, invalid provider reward denom",
-			ccvtypes.NewParams(true, 5, "", "", 5, 1005, "0.5", 1000, 24*21*time.Hour, "0.05", []string{}, []string{"a"}), false,
+			ccvtypes.NewParams(true, 5, "", "", 5, 1005, "0.5", 1000, 24*21*time.Hour, "0.05", []string{}, []string{"a"}, 2*time.Hour), false,
```
Comment (p-offtermatt) [inline x/ccv/consumer/types/params_test.go:74]: Could you add a test with retry delay period = 0? Seems like an obvious edge case to document via test

**Review by p-offtermatt — APPROVED** [review]
LGTM, one nit

File: x/ccv/consumer/types/params_test.go (hunk 1 above)
Comment (smarshall-spitzbart) [reply x/ccv/consumer/types/params_test.go:74]: For consistency we'd prob want to do this for all the non-zero params, but relevant to this PR I've added the test case https://github.com/cosmos/interchain-security/commit/9f305db7a02d1e53c1c0e54f5d5e656febea8e62
